server/document.js

```diff
@@ -182,24 +182,25 @@ export class NextScript extends Component {
     const pagePathname = getPagePathname(pathname)
 
     return <Fragment>
-      {staticMarkup ? null : <script nonce={this.props.nonce} dangerouslySetInnerHTML={{
-        __html: `
-          __NEXT_DATA__ = ${htmlescape(__NEXT_DATA__)}
-          module={}
-          __NEXT_LOADED_PAGES__ = []
-
-          __NEXT_REGISTER_PAGE = function (route, fn) {
-            __NEXT_LOADED_PAGES__.push({ route: route, fn: fn })
-          }${page === '_error' ? `
-
-          __NEXT_REGISTER_PAGE(${htmlescape(pathname)}, function() {
-            var error = new Error('Page does not exist: ${htmlescape(pathname)}')
-            error.statusCode = 404
-
-            return { error: error }
-          })`: ''}
-        `
-      }} />}
+      {staticMarkup ? null : <script
+        id="server-app-state"
+        type="application/json"
+        dangerouslySetInnerHTML={{
+          __html: JSON.stringify({ ...__NEXT_DATA__ }).replace(
+            /<\/script>/g,
+            '%3C/script%3E'
+          ),
+        }}
+      />}
+      <script
```

Comment:
This defeats the purpose of this PR, defines an inline script.

Reply:
As stated above, CSP ignores non-JavaScript <script> tags. Using a <script> tag is the html standard of transmitting JSON as well.

Reply:
But I don't see any `type` attribute set on this `script` tag, so it's treated as javascript. Have you been testing your PR with a strict CSP?